libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp

```diff
@@ -17,7 +22,28 @@ ReturnCode CGroupsParser::parseCGroupsGatewayConfiguration(const json_t *element
         return ReturnCode::FAILURE;
     }
 
-    result.first = settingKey;
-    result.second = settingValue;
+    log_debug() << "key : " << settingKey << " value : " << settingValue;
+
+    if ("memory.limit_in_bytes" == settingKey) {
+        if (m_settings.count(settingKey)) {
+            // if the new value is lexicographical greater than old one set is as the value
+            if (std::stoi(m_settings[settingKey]) < std::stoi(settingValue)) {
+                m_settings[settingKey] = settingValue;
+            }
+        } else {
+            m_settings[settingKey] = settingValue;
+        }
+    } else {
+        log_warning() << "It is possible to apply this rule through control groups of working "
```

Comment:
I think this should be more specific and clear, e.g. "cgroups gateway config \"" <<  config << "\" is not supported by the gateway"